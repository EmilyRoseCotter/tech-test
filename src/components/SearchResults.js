import React from "react";
import "../styles/SearchResults.css";
import PropTypes from "prop-types";

const SearchResults = ({results}) => {
  if (!results.length) {
    return <p className="nr-title">Please enter a valid search term</p>
  } else { 
      return(
        <>
         {results.map((image) => (
           <img className="card-image" src={image} alt="space-image" />
         ))}
       </>
     )
   }
};

SearchResults.propTypes = {
  results: PropTypes.arrayOf(
    PropTypes.string
  ),
};

export default SearchResults;