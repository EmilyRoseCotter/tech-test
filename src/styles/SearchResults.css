.card-image {
  width: 17em;
  margin: auto;
}

.sr-title {
  text-align: center;
  font-size: 20px;
  font-weight: 700;
}

.nr-title {
  text-align: center;
  font-size: 20px;
  font-weight: 700;
}