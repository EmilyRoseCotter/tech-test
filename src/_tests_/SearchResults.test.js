import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  test("renders SearchResult component correctly", () => {
    const { asFragment } = render(<SearchResults/>)

    expect(asFragment()).toMatchSnapshot();
  }) 
})