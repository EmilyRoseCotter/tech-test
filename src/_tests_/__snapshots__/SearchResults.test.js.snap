// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders SearchResult component correctly 1`] = `
<DocumentFragment>
  <img
    alt="space-image"
    class="card-image"
    src="test"
  />
  <img
    alt="space-image"
    class="card-image"
    src="test"
  />
</DocumentFragment>
`;
